Build error: All declarations of 'CKEDITOR_VERSION' must have identical modifiers.

I have just installed and set up ckeditor5-angular following the Getting Started guide. And I am facing the errors below on build:


```

Error: node_modules/@ckeditor/ckeditor5-angular/ckeditor.component.d.ts:8:9 - error TS2687: All declarations of 'CKEDITOR_VERSION' must have identical modifiers.

8         CKEDITOR_VERSION?: string;
          ~~~~~~~~~~~~~~~~


Error: node_modules/@ckeditor/ckeditor5-angular/ckeditor.component.d.ts:8:9 - error TS2717: Subsequent property declarations must have the same type.  Property 'CKEDITOR_VERSION' must be of type '"32.0.0"', but here has type 'string | undefined'.

8         CKEDITOR_VERSION?: string;
          ~~~~~~~~~~~~~~~~

  node_modules/@types/ckeditor__ckeditor5-utils/src/version.d.ts:5:9
    5         CKEDITOR_VERSION: typeof version;
              ~~~~~~~~~~~~~~~~
    'CKEDITOR_VERSION' was also declared here.


Error: node_modules/@types/ckeditor__ckeditor5-utils/src/version.d.ts:5:9 - error TS2687: All declarations of 'CKEDITOR_VERSION' must have identical modifiers.

5         CKEDITOR_VERSION: typeof version;

```


**My versions are:** 

    "@angular/animations": "~13.3.11",
    "@angular/cdk": "^13.3.9",
    "@angular/common": "~13.3.11",
    "@angular/compiler": "~13.3.11",
    "@angular/core": "~13.3.11",
    "@angular/forms": "~13.3.11",
    "@angular/material": "^13.3.9",
    "@angular/material-moment-adapter": "^13.3.9",
    "@angular/platform-browser": "~13.3.11",
    "@angular/platform-browser-dynamic": "~13.3.11",
    "@angular/router": "~13.3.11",
    "@ckeditor/ckeditor5-angular": "^4.0.0",
    "@ckeditor/ckeditor5-build-classic": "^35.4.0",
    "rxjs": "~6.6.0",
    "tslib": "^2.3.0",
    "zone.js": "~0.11.4"

**Temporary fix that works:**

- Go to file \node_modules\@ckeditor\ckeditor5-angular\ckeditor.component.d.ts
- Comment out CKEDITOR_VERSION?: string; on line 8.

--- a/package.json
+++ b/package.json
@@ -27,6 +27,7 @@
     "@angular/platform-browser": "^13.3.11",
     "@angular/platform-browser-dynamic": "^13.3.11",
     "@angular/router": "^13.3.11",
+    "@types/ckeditor__ckeditor5-utils": "^28.0.14",
     "core-js": "^3.21.1",
     "rxjs": "^6.5.5",
     "tslib": "^2.0.3",

--- a/package.json
+++ b/package.json
@@ -27,7 +27,7 @@
     "@angular/platform-browser": "^13.3.11",
     "@angular/platform-browser-dynamic": "^13.3.11",
     "@angular/router": "^13.3.11",
-    "@types/ckeditor__ckeditor5-utils": "^28.0.14",
+    "@types/ckeditor__ckeditor5-utils": ">=28.0.14",
     "core-js": "^3.21.1",
     "rxjs": "^6.5.5",
     "tslib": "^2.0.3",

--- a/src/ckeditor/ckeditor.component.spec.ts
+++ b/src/ckeditor/ckeditor.component.spec.ts
@@ -24,13 +24,13 @@ describe( 'CKEditorComponent', () => {
 		beforeEach( () => {
 			spyOn( console, 'warn' );
 
-			CKEDITOR_VERSION = window.CKEDITOR_VERSION;
+			CKEDITOR_VERSION = ( window as any ).CKEDITOR_VERSION;
 		} );
 
 		afterEach( () => {
 			fixture.destroy();
 
-			window.CKEDITOR_VERSION = CKEDITOR_VERSION;
+			( window as any ).CKEDITOR_VERSION = CKEDITOR_VERSION;
 		} );
 
 		it( 'should create', () => {
@@ -42,7 +42,7 @@ describe( 'CKEditorComponent', () => {
 		} );
 
 		it( 'should print a warning if the "window.CKEDITOR_VERSION" variable is not available', () => {
-			delete window.CKEDITOR_VERSION;
+			delete ( window as any ).CKEDITOR_VERSION;
 
 			fixture = TestBed.createComponent( CKEditorComponent );
 			component = fixture.componentInstance;
@@ -52,7 +52,7 @@ describe( 'CKEditorComponent', () => {
 		} );
 
 		it( 'should print a warning if using CKEditor 5 in version lower than 34', () => {
-			window.CKEDITOR_VERSION = '30.0.0';
+			( window as any ).CKEDITOR_VERSION = '30.0.0';
 
 			fixture = TestBed.createComponent( CKEditorComponent );
 			component = fixture.componentInstance;
@@ -62,7 +62,7 @@ describe( 'CKEditorComponent', () => {
 		} );
 
 		it( 'should not print any warning if using CKEditor 5 in version 34 or higher', () => {
-			window.CKEDITOR_VERSION = '34.0.0';
+			( window as any ).CKEDITOR_VERSION = '34.0.0';
 
 			fixture = TestBed.createComponent( CKEditorComponent );
 			component = fixture.componentInstance;

--- a/src/ckeditor/ckeditor.component.ts
+++ b/src/ckeditor/ckeditor.component.ts
@@ -3,9 +3,11 @@
  * For licensing, see LICENSE.md.
  */
 
+import type { version } from 'ckeditor__ckeditor5-utils';
+
 declare global {
 	interface Window {
-		CKEDITOR_VERSION?: string;
+		CKEDITOR_VERSION: typeof version;
 	}
 }
 

--- a/src/ckeditor/ng-package.json
+++ b/src/ckeditor/ng-package.json
@@ -2,5 +2,9 @@
   "lib": {
     "entryFile": "./index.ts"
   },
+  "allowedNonPeerDependencies": [
+    "@ckeditor/ckeditor5-watchdog",
+    "@types/ckeditor__ckeditor5-utils"
+  ],
   "dest": "../../dist"
 }

--- a/src/ckeditor/package.json
+++ b/src/ckeditor/package.json
@@ -18,11 +18,14 @@
     "ckeditor5",
     "ckeditor 5"
   ],
+  "dependencies": {
+    "@ckeditor/ckeditor5-watchdog": ">=34.0.0",
+    "@types/ckeditor__ckeditor5-utils": "^28.0.14"
+  },
   "peerDependencies": {
     "@angular/core": ">=13.0.0",
     "@angular/common": ">=13.0.0",
     "@angular/forms": ">=13.0.0",
-    "@ckeditor/ckeditor5-watchdog": ">=34.0.0",
     "rxjs": ">=6.0.0"
   },
   "author": "CKSource (http://cksource.com/)",

--- a/src/ckeditor/package.json
+++ b/src/ckeditor/package.json
@@ -20,7 +20,7 @@
   ],
   "dependencies": {
     "@ckeditor/ckeditor5-watchdog": ">=34.0.0",
-    "@types/ckeditor__ckeditor5-utils": "^28.0.14"
+    "@types/ckeditor__ckeditor5-utils": ">=28.0.14"
   },
   "peerDependencies": {
     "@angular/core": ">=13.0.0",

